presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSourceFactory.java

```diff
@@ -82,6 +78,9 @@
     @GuardedBy("this")
     private final List<SettableFuture<LookupSource>> lookupSourceFutures = new ArrayList<>();
 
+    @GuardedBy("this")
+    private final Map<Integer, CompletableFuture<LookupSource>> unspillLookupSourceFuturesByPartitionCache = new HashMap<>();
```

Comment:
what about closing those lookup sources? What happens in case of some abnormal interrupt/cancelling the query?